iam_robolib/src/feedback_controller/joint_impedance_feedback_controller.cpp

```
@@ -37,17 +39,25 @@ void JointImpedanceFeedbackController::initialize_controller(FrankaRobot *robot)
 void JointImpedanceFeedbackController::get_next_step() {}
 
 void JointImpedanceFeedbackController::get_next_step(const franka::RobotState &robot_state,
-                                               TrajectoryGenerator *traj_generator) {
+                                                     TrajectoryGenerator *traj_generator) {
 
   // Read current coriolis terms from model.
   std::array<double, 7> coriolis = model_->coriolis(robot_state);
 
+  JointTrajectoryGenerator* joint_trajectory_generator = dynamic_cast<JointTrajectoryGenerator*>(traj_generator);
+
+  if(joint_trajectory_generator == nullptr) {
+    throw 333;
```

let's throw a valid exception here.